tests/end-to-end/external-solution-integration/pkg/step/cli.go

```diff
@@ -35,6 +40,8 @@ const (
 	CleanupModeOnly CleanupMode = "only"
 	// Execute both steps and cleanup
 	CleanupModeYes CleanupMode = "yes"
+	// Execute steps. If steps fail then run cleanup. Keep resources in case of success
+	CleanupModeOnErrorOnly CleanupMode = "onerroronly"
```

Comment:
Should the `(m *CleanupMode) Set(v string) error` func take care of this as well?